index.html

```diff
@@ -0,0 +1,79 @@
+<!DOCTYPE html>
+<html>
+
+<body>
+    <h1>Genotify</h1>
+    <form>
+        Search for:
+        <br>
+        <input type="text" id="query" placeholder="CDK2, cyclin kinase 2, etc">
+    </form>
+    <br>
+    <button id="searchButton" type="button">Search</button>
+    <p id="basics"></p>
+    <br>
+    <p id="summary"></p>
+    <script>
+    (function() {
```

Comment:
Your code will be executed as soon as it is encountered. Notice the `document.getElementById()` call. It relies on the fact that the DOM is ready, so by immediately making this call, there's a change that it returns `null` if the page hasn't be parsed yet. This is fine for your current setup (placing the script at the end of the body), as all the html will be parsed before the JavaScript is reached. However, you'll likely segregate your JS code into their own files in the future, and in doing so, you'll encounter this issue.

Look into this: https://developer.mozilla.org/en-US/docs/Web/Events/DOMContentLoaded